From 2de138957ea0ea05675f8d1f2030fa0b0fef6910 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Fri, 28 Jul 2017 11:22:31 +0800
Subject: [PATCH 2246/5242] MLK-16095-01 i2c: mux: pca954x: add i2c bus switch
 PCA9646 chip support

commit  217d492b137f1810f4d9e1f3bc63c9b547a29c15 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add i2c bus switch PCA9646 chip support, which 2-wire bus switch
and buffered 4-channel.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/i2c/i2c-mux-pca954x.txt    |    1 +
 drivers/i2c/muxes/i2c-mux-pca954x.c                |    7 +++++++
 2 files changed, 8 insertions(+)

diff --git a/Documentation/devicetree/bindings/i2c/i2c-mux-pca954x.txt b/Documentation/devicetree/bindings/i2c/i2c-mux-pca954x.txt
index 34d9150..37da062 100644
--- a/Documentation/devicetree/bindings/i2c/i2c-mux-pca954x.txt
+++ b/Documentation/devicetree/bindings/i2c/i2c-mux-pca954x.txt
@@ -14,6 +14,7 @@ Required Properties:
     "nxp,pca9547", "nxp,pca9847",
     "nxp,pca9548", "nxp,pca9848",
     "nxp,pca9849"
+    "nxp,pca9646"
 
   - reg: The I2C address of the device.
 
diff --git a/drivers/i2c/muxes/i2c-mux-pca954x.c b/drivers/i2c/muxes/i2c-mux-pca954x.c
index fbc7480..311464b 100644
--- a/drivers/i2c/muxes/i2c-mux-pca954x.c
+++ b/drivers/i2c/muxes/i2c-mux-pca954x.c
@@ -68,6 +68,7 @@ enum pca_type {
 	pca_9847,
 	pca_9848,
 	pca_9849,
+	pca_9646,
 };
 
 struct chip_desc {
@@ -177,6 +178,10 @@ struct pca954x {
 			.part_id = 0x109,
 		},
 	},
+	[pca_9646] = {
+		.nchans = 4,
+		.muxtype = pca954x_isswi,
+	},
 };
 
 static const struct i2c_device_id pca954x_id[] = {
@@ -192,6 +197,7 @@ struct pca954x {
 	{ "pca9847", pca_9847 },
 	{ "pca9848", pca_9848 },
 	{ "pca9849", pca_9849 },
+	{ "pca9646", pca_9646 },
 	{ }
 };
 MODULE_DEVICE_TABLE(i2c, pca954x_id);
@@ -210,6 +216,7 @@ struct pca954x {
 	{ .compatible = "nxp,pca9847", .data = &chips[pca_9847] },
 	{ .compatible = "nxp,pca9848", .data = &chips[pca_9848] },
 	{ .compatible = "nxp,pca9849", .data = &chips[pca_9849] },
+	{ .compatible = "nxp,pca9646", .data = &chips[pca_9646] },
 	{}
 };
 MODULE_DEVICE_TABLE(of, pca954x_of_match);
-- 
1.7.9.5

